tests/test_representation_manager.hh

```diff
@@ -151,6 +151,61 @@ namespace rascal {
                                   {"normalize", true}}};
   };
 
+  struct MultipleStructureSOAPCenterMask
```

Comment:
It seems a bit ridiculous to have to make a new one of these representation classes, with the same boilerplate code, every time we want to try out a feature of the (completely unrelated!) structure manager. I'm not suggesting we do it in this PR, but let's refactor the test code soon; this is getting out of hand.

Reply:
We started by making the tests avoiding such code repetition. In the end it made the tests very hard to understand so I moved to a more verbose organization that avoid repeating the functional code but states all the relevant inputs in 1 structure (avoiding scattering them over).
Here it is repeated but inherit from a different neighbourlist definition.
To avoid repretition the inheritance is templated now.